### PR TITLE
Update `@typescript-eslint/typescript-estree` to v3.7.1

### PR DESCRIPTION
**Pull request by fisker**
[`typescript-eslint/typescript-estree@v3.7.0` release note](https://github.com/typescript-eslint/typescript-eslint/releases)

Support type annotations on catch clauses.

Logical assignment seems requires `typescript@4`

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
`TSNullKeyword` seems removed in https://github.com/typescript-eslint/typescript-eslint/commit/cbaac775c66fc74277eeecd2a6f9fcff2a3329c6

`node.type === "Literal" && node.value === null` should be enough for it.

But we always use `isXXXLiteral` to test `Literal`, so I add `isNullLiteral` and also check `NullLiteral`(babel).

File: src/language-js/printer-estree.js [hunk 1]
```diff
@@ -4997,7 +4997,7 @@ function shouldHugType(node) {
         n.type === "VoidTypeAnnotation" ||
         n.type === "TSVoidKeyword" ||
         n.type === "NullLiteralTypeAnnotation" ||
-        n.type === "TSNullKeyword"
```
Comment (thorn0) [inline src/language-js/printer-estree.js:5000]: `babel-ts` generates `TSNullKeyword` nodes. Also are we sure these changes are expected? I don't see anything about this in the changelog of `typescript-estree`.

Comment (fisker) [reply src/language-js/printer-estree.js:5000]: I didn't test locally, waiting for CI result.

This change didn't mentioned, but tests failed. https://github.com/prettier/prettier/runs/892776097#step:7:1263

#8806 also failed.

Comment (fisker) [reply src/language-js/printer-estree.js:5000]: You are right, `TSNullKeyword` can't be removed. Restored.

Comment (thorn0) [reply src/language-js/printer-estree.js:5000]: @bradzacher It's not expected that the AST format changed for the `null` type literal (TS 3.9.7), is it?

Comment (bradzacher) [reply src/language-js/printer-estree.js:5000]: No, but I'm just now realising that I probably introduced a regression for <4.0.0 with the latest version. I'll submit a fix to undo it.

Comment (fisker) [reply src/language-js/printer-estree.js:5000]: Will it  change for typescript 4?

Comment (fisker) [reply src/language-js/printer-estree.js:5000]: Thank you!

**Comment by fisker** [comment]
AST change reverted, Waiting for `@typescript-eslint/typescript-estree` new release.

**Comment by bradzacher** [comment]
you can test this against the `canary` version that'll be published in a few minutes.
full release will be Monday morning PST

**Comment by fisker** [comment]
Read to merge.

**Pull request merged**